freqtrade/optimize/hyperopt.py

```diff
@@ -288,6 +288,9 @@ def start(self) -> None:
                             'total_tries': self.total_tries,
                             'result': f_val[j]['result'],
                         })
+                        logger.debug(f"Optimizer params: {f_val[j]['params']}")
+                    for j in range(cpus):
+                        logger.debug(f"Opimizer state: Xi: {opt.Xi[-j-1]}, yi: {opt.yi[-j-1]}")
```

Comment:
Maybe i'm missing something - but don't the 2 lines print the same thing in different formats (i prefer the "Opitmizer params:" one ...)?

Trying this - all values match (if you take only the values of the dict).


```
.2019-04-23 19:37:03,328 - freqtrade.optimize.hyperopt - DEBUG - Optimizer params: {'mfi-value': 12, 'fastd-value': 28, 'adx-value': 38, 'rsi-value': 37, 'mfi-enabled': False, 'fastd-enabled': False, 'adx-enabled': False, 'rsi-enabled': True, 'trigger': 'macd_cross_signal', 'sell-mfi-value': 76, 'sell-fastd-value': 84, 'sell-adx-value': 92, 'sell-rsi-value': 97, 'sell-mfi-enabled': True, 'sell-fastd-enabled': True, 'sell-adx-enabled': False, 'sell-rsi-enabled': True, 'sell-trigger': 'sell-bb_upper', 'roi_t1': 102, 'roi_t2': 29, 'roi_t3': 24, 'roi_p1': 0.03907810596868612, 'roi_p2': 0.017418899486970653, 'roi_p3': 0.1462967283445416, 'stoploss': -0.37168748310135113}

2019-04-23 19:37:03,328 - freqtrade.optimize.hyperopt - DEBUG - Opimizer state: Xi: [12, 28, 38, 37, False, False, False, True, 'macd_cross_signal', 76, 84, 92, 97, True, True, False, True, 'sell-bb_upper', 102, 29, 24, 0.03907810596868612, 0.017418899486970653, 0.1462967283445416, -0.37168748310135113], yi: 2.2217194636218585
```

The sequence of printing is exactly reversed ... but still there is no added value in printing it twice i think... (unless i am not seeing something )

Reply:
they are not the same: params contains names of the hyperopt space dimensions, while Xi does not; in the same time, second line contains more or less complete optimizer state: Xi and yi. There is also so called models and hessians, etc. but they require lot's of space on the screen and in the logs, so not included.
Yes, I can try to print them in the same order (they come as you see in different order, that's why I print them in two diff. loops). That's one possibility.
These are values that come from different sources: params is produced directly by our generate_optimizer() method, while xi and yi come as parts of the state of the optimizer itself, i.e. from internals of the optimizer. So both values can be usefull for debugging (these are DEBUG messages, notice it) -- if they differ that means that something goes wrong with passing values to the optimizer...
Another possibility is to print params (with names of the space dimensions) and print yi from the internals of the optimizer.
Third possibility (I like it after thinking on it): print 'params' from generate_optimizer() AND (additionally to the current implementation) print there the 'loss' also from generate_optimizer() -- that will be one message. Plus, print xi and yi from internal state of the optimizer. So that one could compare them if he need it for some reasons.